Makefile

```diff
@@ -102,19 +147,65 @@ leak: LDFLAGS += -fsanitize=leak
 leak: clean $(NAME)
```

Comment:
Why is the  target leak defined multiple times instead of just once and then have the different steps executed one after the other?

Reply:
They are target-specific variable values (see [GNU Make manual 6.11](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html#:~:text=Target%2Dspecific%20variables%20have%20the,variable%20value%20to%20be%20preferred.))

The intent is to (re)define a variable value for a specific target.

For example for `leak` we want to have -fsanitize=leak in CFLAGS and LDFLAGS. We can't (for technical reasons) redefine and have prerequisite on the same line. Therefore a line sets the value for CFLAGS and LDFLAGS respectively. The third line then executes `clean` and `$(NAME)`. (This is also the way how they show it in the manual)
For clarity's sake: Target specific values are also effective for targets called in the recipe. So if I call `$(NAME)` in `leak` CFLAGS will have another value

Reply:
Very interesting concept, learned sth new